packages/create-amplify/src/package-manager-controller/package_manager.ts

```diff
@@ -0,0 +1,77 @@
+export type PackageManagerName =
+  | 'npm'
+  | 'yarn-classic'
+  | 'yarn-modern'
+  | 'pnpm';
+type PackageManagerExecutable = 'npm' | 'yarn' | 'pnpm';
+type PackageManagerBinaryRunner = 'npx' | 'yarn' | 'pnpm';
+type PackageManagerInstallCommand = 'install' | 'add';
+type PackageManagerLockFile =
+  | 'package-lock.json'
+  | 'yarn.lock'
+  | 'pnpm-lock.yaml';
+type PackageManagerInitDefault = Readonly<string[]>;
+export type PackageManagerProps = {
+  name: PackageManagerName;
+  executable: PackageManagerExecutable;
+  binaryRunner: PackageManagerBinaryRunner;
+  installCommand: PackageManagerInstallCommand;
+  lockFile: PackageManagerLockFile;
+  initDefault: PackageManagerInitDefault;
+};
+
+export type PackageManagers = {
+  [key in PackageManagerName]: PackageManagerProps;
+};
+
+/**
+ *
+ */
+export class PackageManagerBase {
+  private packageManagers: PackageManagers = {
+    npm: {
```

Comment:
`Base` suffix is usually reserved for abstract classes that are root of class hierarchy. If we don't `extend PackageManagerBase` then this naming is not good.

However it seems that this class should be `PackageManagerFactory` since it has `getPackageManager`.
But I wouldn't leave `getPackageManager` in current form here but refactor the code in such a way that all details related to how to create `PackageManagerController` (and `Initializer` if we don't merge it) should come here.

If we don't merge "controller" with "initializer" I think the api of this class should be:
```
export class PackageManagerFactory {

getPackageManagerController(userAgent): PackageManagerController;

getProjectInititlizer(userAgent): ProjectInitializer;

}
```

or some variation of that.

Reply:
Done